senderworker/src/main/scala/com/gu/notifications/ec2worker/SenderWorker.scala

```diff
@@ -0,0 +1,87 @@
+import scala.jdk.CollectionConverters._
+import org.slf4j.LoggerFactory
+import org.slf4j.Logger
+import _root_.models.{Android, AndroidBeta, AndroidEdition, Ios, IosEdition, Platform}
+import com.gu.notifications.worker.IOSSender
+import com.gu.notifications.worker.AndroidSender
+
+object SenderWorker extends App {
+
+  implicit val logger: Logger = LoggerFactory.getLogger(this.getClass)
+
+  val notification = BreakingNewsNotification(
+    id = UUID.randomUUID(),
+    `type` = NotificationType.BreakingNews,
+    title = Some("French president Francois Hollande says killers of Normandy priest claimed to be from Islamic State"),
+    message = Some("French president Francois Hollande says killers of Normandy priest claimed to be from Islamic State"),
+    thumbnailUrl = None,
+    sender = "matt.wells@guardian.co.uk",
+    link = Link.Internal(
+      "world/2016/jul/26/men-hostages-french-church-police-normandy-saint-etienne-du-rouvray",
+      Some("https://gu.com/p/4p7xt"),
+      GITContent,None,
+    ),
+    imageUrl = None,
+    importance = Importance.Major,
+    topic = List(Topic(TopicTypes.Breaking, "uk"), Topic(TopicTypes.Breaking, "us"), Topic(TopicTypes.Breaking, "au"), Topic(TopicTypes.Breaking, "international")),
+    dryRun = Some(true)
+  )
+
+  val tokens = ChunkedTokens(
+    notification = notification,
+    range = ShardRange(0, 1),
+    tokens = List("token")
+  )
+
+  val sqsEvent: SQSEvent = {
+    val event = new SQSEvent()
+    val sqsMessage = new SQSMessage()
+    sqsMessage.setBody(Json.stringify(Json.toJson(tokens)))
+    sqsMessage.setAttributes((Map("SentTimestamp" -> s"${Instant.now.toEpochMilli}").asJava))
+    event.setRecords(List(sqsMessage).asJava)
+    event
+  }
+
+  println("Sender workers start")
+  logger.info("Sender workers start")
+  val config = Configuration.fetchConfiguration()
+
+  logger.info("Sender worker - Ios started")
+  val iosSender = new IOSSender(Configuration.fetchApns(config, Ios))
+  iosSender.handleChunkTokens(sqsEvent, null)
+  
+  logger.info("Sender worker - Android started")
+  val androidSender = new AndroidSender(Configuration.fetchFirebase(config, Android), Some(Android.toString()))
+  androidSender.handleChunkTokens(sqsEvent, null)
+
+  logger.info("Sender worker - IosEdition started")
+  val iosEditionSender = new IOSSender(Configuration.fetchApns(config, IosEdition))
+  iosEditionSender.handleChunkTokens(sqsEvent, null)
+
+  logger.info("Sender worker - AndroidBeta started")
+  val androidBetaSender = new AndroidSender(Configuration.fetchFirebase(config, AndroidBeta), Some(AndroidBeta.toString()))
+  androidBetaSender.handleChunkTokens(sqsEvent, null)
+
+  logger.info("Sender worker - AndroidEdition started")
+  val androidEditionSender = new AndroidSender(Configuration.fetchFirebase(config, AndroidEdition), Some(AndroidEdition.toString()))
+  androidEditionSender.handleChunkTokens(sqsEvent, null)
+
+  logger.info("Sender worker all started")
+  Thread.sleep(60*60*1000)
```

Comment:
For now, when the servers start we should expect to see in the logs these messages once? And then the process terminates/waits, but doesn't continue to `handleChunkTokens`? Maybe I've misunderstood the implementation

Reply:
I think this raises an interesting point, which may be worth testing/experimenting/validating. If an error is thrown in the loop that listens to the sqs queue, is the service automatically restarted, and yes after how long? I had a previous experience where unhandled errors when listening to the queue terminated the server but there was no auto-restart process (server remained dead)

Reply:
In my first version where the service just quits after calling the sender stuff, it gets started up every minute.  But I also recall that I saw once the process was not started up again after some time.  So I suspect that we may not be able to rely on this auto-restart process completely, but need to implement some kind of [health check](https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-health-checks.html#instance-health-detection) for the autoscaling group to start up a new EC2 instance when the process does not work properly.

One idea may be that the service sends some kinds of cloudwatch metric regularly inside the loop.  When the metric is missing for a certain period, the health check fails.

I have created a ticket to look into this. LIVE-4744](https://theguardian.atlassian.net/browse/LIVE-4744)